xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/templates/comments2.vm

```diff
@@ -21,10 +21,34 @@
 #set($showcomments = $xwiki.getSpacePreference('showcomments'))
 #end
 #if($showcomments!="no")
-#if($xwiki.getSpacePreferenceAsInt('commentsorder', 1) == 0)
 #set($comments = $doc.getComments())
+#set ($commentMap = {})
+#set ($keysWithMultipleComments = $collectiontool.getSet())
+#foreach ($comment in $comments)
+#set ($key = $comment.date)
+#if ($commentMap.containsKey($key))
+#set ($discard = $commentMap.get($key).add($comment))
+#set ($discard = $keysWithMultipleComments.add($key))
 #else
-#set($comments = $doc.getComments(false))
+#set ($discard = $commentMap.put($key, [$comment]))
+#end
+#end
+#set ($orderedKeys = $collectiontool.getLinkedList())
+#set ($discard = $orderedKeys.addAll($commentMap.keySet()))
+#set ($discard = $collectiontool.sortModifiable($orderedKeys))
+#if($xwiki.getSpacePreferenceAsInt('commentsorder', 1) == 1)
+#set ($orderSuffix = ":asc")
```

Comment:
If there is no code to evaluate, we can use `'` instead of `"`, right?

Reply:
Good call, I need to fix it too in comments.vm.